railties/lib/rails/generators/rails/scaffold_controller/templates/controller.rb

```diff
@@ -5,11 +5,23 @@
 <% module_namespacing do -%>
 class <%= controller_class_name %>Controller < ApplicationController
   before_action :set_<%= singular_table_name %>, only: [:show, :edit, :update, :destroy]
+  
+  <%- if collection_routing? -%>
+  # GET <%= route_url %>/1..10,17
+  def index
+    @<%= plural_table_name %> = <%= singular_table_name %>_ids ? <%= orm_class.find(class_name, "#{singular_table_name}_ids") %> : <%= orm_class.all(class_name) %>
```

Comment:
I think this ternary operator needs to be changed at least to `<%= singular_table_name %>_ids.any?`.

In other words, the generated method should change from:

```ruby
  def index
    @posts = post_ids ? Post.find(post_ids) : Post.all
  end
```

to

```ruby
  def index
    @posts = post_ids.any? ? Post.find(post_ids) : Post.all
  end
```

that is because `post_ids` returns an empty hash when you simply browse to `/posts`.

Alternatively, you can keep the `post_ids` in the ternary operator, but make sure that the `post_ids` method returns some truthy or falsey value accordingly.